### PR TITLE
Add optional support for CmdS3Copy

### PR DESCRIPTION
**Pull request by dbradf**
This feature is needed for some of the work I'm doing to convert all resmoke tasks to use generate.tasks.

### CODE REVIEW

File: tests/shrub/test_operations.py [hunk 1]
```diff
@@ -592,11 +592,28 @@ def test_parameters(self):
...
+        c = CmdS3Copy()
+        f0 = AwsCopyFile() \
+            .display_name("f0 name") \
+            .source("bucket 0", "path 0") \
+            .destination("bucket 1", "path 1")
+        f1 = AwsCopyFile() \
+            .display_name("f1 name") \
+            .source("bucket 2", "path 2") \
+            .destination("bucket 3", "path 3")
+        c.aws_key("aws key").optional().aws_secret("aws secret")\
```
Comment (jimoleary) [inline tests/shrub/test_operations.py]: For some reason the placement of the optional call made me think this case making the remaining builder invocations optional.

This is surely my mistake but for clarity maybe move optional() to the end of the chain.

File: tests/shrub/test_operations.py [hunk 2]
```diff
@@ -592,11 +592,28 @@ def test_parameters(self):
...
+            .file(f0).files([f1])
+
+        p = params(c)
+        assert p["optional"]
```
Comment (jimoleary) [inline tests/shrub/test_operations.py]: do you need a separate __assert "optional" in p__ to balance out line 595.

File: tests/shrub/test_operations.py (hunk 1 above)
Comment (dbradf) [reply tests/shrub/test_operations.py]: done

File: tests/shrub/test_operations.py (hunk 2 above)
Comment (dbradf) [reply tests/shrub/test_operations.py]: done

**Pull request merged**